java-snapshot-testing-core/src/main/java/au/com/origin/snapshots/Snapshot.java

```diff
@@ -159,15 +163,21 @@ public void toMatchSnapshot() {
                     throw new IllegalStateException("No compatible reporters found for comparator " + comparator);
                 }
 
-                for (SnapshotDiffReporter reporter : reporters) {
-                    reporter.reportDiff(getSnapshotName(), rawSnapshot, currentObject);
+                for (SnapshotReporter reporter : reporters) {
+                    reporter.report(getSnapshotName(), rawSnapshot, currentObject);
                 }
 
                 throw new SnapshotMatchException("Error matching snapshot");
             }
         } else {
-            // Create New Snapshot
-            snapshotFile.push(currentObject);
+            if (this.isCi) {
+                log.warn("We detected you are running on a CI Server - if this is incorrect please override the isCI() method in SnapshotConfig");
```

Comment:
Just thinking maybe this should be `log.error`

Reply:
I'm not sure though - warning sounds right but could be missed if their CI logging is only error or something.  I'll leave it up to you to decide cause I can find arguments for both ways.

Reply:
I went back and forth on this. I also think log.error would be better in this case. Will change it.